google/cloud/bigtable/table.h

```diff
@@ -482,7 +482,7 @@ class Table {
    * @par Example
    * @snippet data_async_snippets.cc async check and mutate
    */
-  future<StatusOr<google::bigtable::v2::CheckAndMutateRowResponse>>
+  future<StatusOr<bool>>
```

Comment:
`StatusOr<bool>` is easy to use incorrectly (mistaking `if (b)` for `if (b && *b)`), so perhaps we should avoid it.

Reply:
Yeah.... That would require changing the non-Async version too, maybe with a `kPredicateNotMatched` and `kPredicateMatched` enum.  We can either fix both in this PR, or I can create a separate PR to fix both after this one.

I slightly prefer the second option for future software archaeologists.

/cc: @devjgm because he was interested in this.

Reply:
> I slightly prefer the second option for future software archaeologists.

Sounds good to me.